src/util.ts

```diff
@@ -13,6 +14,289 @@ class Util {
             character: character
         } as Location;
     }
+
+    public identifyPlatform(platformCodeName: string): RokuHardwarePlatform {
+        const rokuPlatforms = this.getAllRokuPlatforms();
+        // Try an exact match
+        return rokuPlatforms.find(p => p.codeName.toLowerCase() === platformCodeName.toLowerCase()) ??
+            // Try a partial match
+            rokuPlatforms.find(p => p.codeName.toLowerCase().includes(platformCodeName.toLowerCase())) ??
+            // Split the code name and try an exact match with each part
+            rokuPlatforms.find(
+                p => platformCodeName.split(/;|,|_/) // Split on semicolon, comma, and underscore
+                    .filter(part => part.toLowerCase() !== '4k') // Ignore 4K indicator
+                    .some(part => p.codeName.toLowerCase().includes(part.toLowerCase()))
+            ) ??
+            // Return Unknown
+            {
+                productName: 'Unknown',
+                codeName: platformCodeName,
+                model: 'Unknown',
+                productionStatus: ProductionStatus.Unknown
+            };
+    }
+
+    // TODO: where to put this? should we store the codeName as string instead of RokuHardwarePlatform?
+    /**
+     * @see https://developer.roku.com/docs/specs/hardware.md
+    */
+    public getAllRokuPlatforms(): RokuHardwarePlatform[] {
+        return [
+            // Current Roku models
+            // The following models are currently being manufactured and are supported:
+            {
+                productName: 'Roku Express',
+                codeName: 'Nemo',
+                model: '3930X',
+                productionStatus: ProductionStatus.Current
```

Comment:
Is there a benefit to us storing the end-of-life date on this rather than a hardcoded production status? Just thinking out loud, these statuses will need to be manually changed in this package as the years progress. Perhaps that wouldn't get us any benefit?

Reply:
Nevermind